packages/runner/src/providers/container/index.ts

```diff
@@ -0,0 +1,219 @@
+import { Docker } from "node-docker-api"
+import fetch from 'node-fetch'
+import { ReadStream } from 'fs'
+import API from '@mesg/api/lib/lcd'
+import * as ServiceType from '@mesg/api/lib/typedef/service';
+import { IService } from "@mesg/api/lib/service-lcd"
+import { IRunner } from '@mesg/api/lib/runner-lcd'
+import { findHash } from "@mesg/api/lib/util/txevent"
+import { Provider } from '../../index'
+import { Network } from "node-docker-api/lib/network"
+import Container from './container'
+import { createHash } from 'crypto'
+import { IAccount } from "@mesg/api/lib/account-lcd";
+
+const debug = require('debug')('runner')
+
+const ENGINE_NETWORK_NAME = 'engine'
+const PREFIX = 'mesg_srv_'
+
+type Labels = { [key: string]: string }
+
+export default class DockerContainer implements Provider {
+
+  private _api: API
+  private _client: Docker
+  private _mnemonic: string
+  private ipfsGateway = "http://ipfs.app.mesg.com:8080/ipfs"
+  private serviceEndpoint: string
+
+  constructor(endpoint: string, mnemonic: string, serviceEndpoint: string = ENGINE_NETWORK_NAME) {
+    this._api = new API(endpoint)
+    this._client = new Docker(null)
+    this._mnemonic = mnemonic
+    this.serviceEndpoint = serviceEndpoint
+  }
+
+  async start(serviceHash: string, env: string[]): Promise<IRunner> {
+
+    debug('fetch service')
+    const service = await this._api.service.get(serviceHash)
+    const account = await this._api.account.import(this._mnemonic)
+    const { runnerHash, instanceHash, envHash } = await this._api.runner.hash(account.address, service.hash, env)
+    debug('broadcast tx')
+
+    const labels = {
+      'mesg.service': service.hash,
+      'mesg.runner': runnerHash,
+    }
+
+    const image = await this.build(service)
+
+    let serviceNetwork = await this.findNetwork(labels)
+    if (!serviceNetwork) serviceNetwork = await this._client.network.create({
+      name: PREFIX + service.hash,
+      labels
+    })
+
+    const engineNetwork = await this.findNetwork({ 'mesg.engine': 'true' })
+    if (!engineNetwork) throw new Error('engine network not found')
+
+    for (const dep of service.dependencies || []) {
+      const container = new Container({
+        Args: dep.args,
+        Command: dep.command,
+        Env: dep.env,
+        Image: dep.image,
+        Labels: {
+          ...labels,
+          'mesg.dependency': dep.key,
+        },
+        HostConfig: {
+          Mounts: this.convertVolumes(service, dep.volumes, dep.volumesFrom, dep)
+        }
+      }, PREFIX + service.hash + '_' + dep.key)
+      container.addPorts(dep.ports)
+      container.connectTo(serviceNetwork, ['service'])
```

Comment:
wrong alias for the dependencies:
```suggestion
      container.connectTo(serviceNetwork, [dep.key])
```